### PR TITLE
Add toolchain artifacts GH workflow

### PR DESCRIPTION
**Pull request by emostov**

### CODE REVIEW

File: .github/workflows/artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch: # Allows manual invocation
+
+jobs:
+  build:
+    name: Build Toolchain Artifacts
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
```
Comment (r-n-o) [inline .github/workflows/artifacts.yml:14]: woah there! Really? 2 hours? 😮

File: .github/workflows/artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Download submodules
+        run: git submodule update --init --recursive
+        shell: bash
+
+      - name: Pull lfs artifacts
+        run: git lfs pull
+        shell: bash
+
+      - name: Run `make dist`
+        shell: 'script -q -e -c "bash {0}"'
```
Comment (r-n-o) [inline .github/workflows/artifacts.yml:32]: I didn't know what `script` was so I looked it up. The `man` description thinks it's a tool for students! TIL. 
```
The script utility makes a typescript of everything printed on your terminal. 
It is useful for students who need a hardcopy record of an interactive session as
proof of an assignment, as the typescript file can be printed out later with lpr(1).
```

File: .github/workflows/artifacts.yml (hunk 1 above)
Comment (emostov) [reply .github/workflows/artifacts.yml:14]: It consistently takes ~55 minutes. Not much I can do about that right now, but I talked to @lrvick and I think we should have some performance improvements in the pipeline

Comment (r-n-o) [reply .github/workflows/artifacts.yml:14]: Dang. Ok then! 2hrs is a reasonable timeout 😬  (for now)

File: .github/workflows/artifacts.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        run: make dist
```
Comment (james-callahan) [inline .github/workflows/artifacts.yml:33]: Does this need `PRESERVE_CACHE=true`

**Pull request merged**